src/signals-mach.c

```diff
@@ -588,6 +588,16 @@ void *mach_profile_listener(void *arg)
 #else
                 bt_size_cur += rec_backtrace_ctx((jl_bt_element_t*)bt_data_prof + bt_size_cur, bt_size_max - bt_size_cur - 1, uc, NULL);
 #endif
+                jl_ptls_t ptls = jl_all_tls_states[i];
+
+                // store threadid but add 1 as 0 is preserved to indicate end of block
+                bt_data_prof[bt_size_cur++].uintptr = ptls->tid + 1;
+
+                // store task id
+                bt_data_prof[bt_size_cur++].uintptr = ptls->current_task;
+
+                // store cpu cycle clock
+                bt_data_prof[bt_size_cur++].uintptr = cycleclock();
```

Comment:
Are they always inbound? Do we need to update `jl_profile_is_buffer_full` so that there are always four additional slots available?

https://github.com/JuliaLang/julia/blob/2ebbb2b3a0b54eae66549bd058ae334cb3642e50/src/signal-handling.c#L38-L42

i.e., `+ 4` instead of `+ 1`

Reply:
good point! no, I'll update